server/src/main/java/org/apache/iotdb/db/tools/TsFileSplitTool.java

```diff
@@ -0,0 +1,257 @@
+import org.apache.iotdb.tsfile.file.MetaMarker;
+import org.apache.iotdb.tsfile.file.header.ChunkHeader;
+import org.apache.iotdb.tsfile.file.header.PageHeader;
+import org.apache.iotdb.tsfile.file.metadata.ChunkMetadata;
+import org.apache.iotdb.tsfile.file.metadata.TimeseriesMetadata;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.iotdb.tsfile.fileSystem.FSFactoryProducer;
+import org.apache.iotdb.tsfile.read.TsFileSequenceReader;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.reader.page.PageReader;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.write.chunk.ChunkWriterImpl;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+import org.apache.iotdb.tsfile.write.writer.TsFileIOWriter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TsFileSplitTool {
+
+  private static final Logger logger = LoggerFactory.getLogger(TsFileSplitTool.class);
+
+  private final String filename;
+
+  /**
+   * If the chunk point num is lower than this threshold, it will be deserialized into points,
+   * default is 100
+   */
+  private final long chunkPointNumLowerBoundInCompaction =
+      IoTDBDescriptor.getInstance().getConfig().getChunkPointNumLowerBoundInCompaction();
+
+  private final double targetSplitFileSize =
+      IoTDBDescriptor.getInstance().getConfig().getTargetCompactionFileSize();
+
+  /** Maximum index of plans executed within this TsFile. */
+  protected long maxPlanIndex = Long.MIN_VALUE;
+
+  /** Minimum index of plans executed within this TsFile. */
+  protected long minPlanIndex = Long.MAX_VALUE;
+
+  public static void main(String[] args) throws IOException {
+    String fileName = args[0];
+    logger.info("Splitting TsFile {} ...", fileName);
+    new TsFileSplitTool(fileName).run();
+  }
+
+  /* construct TsFileSketchTool */
+  public TsFileSplitTool(String filename) {
+    this.filename = filename;
+  }
+
+  /* entry of tool */
+  @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
+  public void run() throws IOException {
+    TsFileIOWriter writer = null;
+
+    try (TsFileSequenceReader reader = new TsFileSequenceReader(filename)) {
+      Iterator<List<Path>> pathIterator = reader.getPathsIterator();
+      Set<String> devices = new HashSet<>();
+      String[] filePathSplit = filename.split(IoTDBConstant.FILE_NAME_SEPARATOR);
+      int versionIndex = Integer.parseInt(filePathSplit[filePathSplit.length - 3]) + 1;
+      // to avoid compaction after restarting. NOTICE: This will take effect only in
+      filePathSplit[filePathSplit.length - 2] = "10";
```

Comment:
Make the "10" a input parameter is better :D

Reply:
Fixed, I also added a default num as a global param `private static final String defaultLevelNum = "10"`